### PR TITLE
[fix] Add error handling for runs upgrade CLI

### PR DESCRIPTION
**Pull request by alberttorosyan**

### CODE REVIEW

**Comment by alberttorosyan** [comment]
resolves CLI issue related to #1685

File: aim/cli/storage/commands.py [hunk 1]
```diff
@@ -55,22 +55,32 @@ def to_3_11(ctx, hashes):
...
+                run.update_metrics()
+            else:
+                click.echo(f'Run {run.hash} is already up to date. Skipping')
+        except Exception:
+            click.echo(f'\nFailed to optimize metrics for run \'{run_hash}\'. Skipping.', err=True)
+            remaining_runs.append(run_hash)
+
+    if not remaining_runs:
+        click.echo('Finished optimizing metric data. '
+                   'In case of any issues the following command can be used to restore data:')
```
Comment (mihran113) [inline aim/cli/storage/commands.py:79]: Maybe makes sense to output this message in any case?

**Review by mihran113 — COMMENTED** [review]
some thoughts regarding to the messages

File: aim/cli/storage/commands.py [hunk 2]
```diff
@@ -55,22 +55,32 @@ def to_3_11(ctx, hashes):
...
-    click.echo('Finished optimizing metric data. '
-               'In case of any issues the following command can be used to restore data:')
-    click.secho(f'aim storage --repo {repo.root_path} restore \'*\'', fg='yellow')
+        try:
+            run = MaintenanceRun(run_hash, repo=repo)
+            if run.check_metrics_version():
+                backup_run(run)
+                run.update_metrics()
+            else:
+                click.echo(f'Run {run.hash} is already up to date. Skipping')
```
Comment (mihran113) [inline aim/cli/storage/commands.py:72]: Do we want output errors for each run though? Cause in the end there's a message anyway informing that some runs were skipped.

File: aim/cli/storage/commands.py (hunk 1 above)
Comment (alberttorosyan) [reply aim/cli/storage/commands.py:79]: fair point, will change

**Review by mihran113 — APPROVED** [review]

**Pull request merged**